dotnet/src/SemanticKernel.Abstractions/Filters/Prompt/PromptFilterCollection.cs

```diff
@@ -0,0 +1,142 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft.SemanticKernel;
+
+/// <summary>
+/// Provides a collection of <see cref="IPromptFilter"/> instances.
+/// </summary>
+[Experimental("SKEXP0004")]
+internal sealed class PromptFilterCollection : IList<IPromptFilter>, IReadOnlyList<IPromptFilter>
```

Comment:
We could just have a:
```C#
internal sealed class NonNullItemsCollection<T> : IList<T>, IReadOnlyList<T>
{
   ...
}
```
rather than having both FunctionFilterCollection and PromptFilterCollection.

Reply:
Makes sense, thank you. I added generic collection for non-null items.